File: Node/models/Marks.js
const mongoose = require("mongoose");

const subjectSchema = new mongoose.Schema({
  marks: Number,
  credit: Number,
});

const subjectMarksSchema = new mongoose.Schema({
  subject1: subjectSchema,
  subject2: subjectSchema,
  subject3: subjectSchema,
  subject4: subjectSchema,
  subject5: subjectSchema,
});

const marksSchema = new mongoose.Schema({
  uid: { type: String, required: true },
  semester: {
    sem1: subjectMarksSchema,
    sem2: subjectMarksSchema,
    sem3: subjectMarksSchema,
    sem4: subjectMarksSchema,
    sem5: subjectMarksSchema,
    sem6: subjectMarksSchema,
    sem7: subjectMarksSchema,
    sem8: subjectMarksSchema,
  }
});

module.exports = mongoose.models.Marks || mongoose.model("Marks", marksSchema);

File: Node/studentapi/studentapi/src/main/java/com/adobe/studentapi/model/Marks.java
package com.adobe.studentapi.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document("marks")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Marks {
    @Id
    private String id;

    @Indexed(unique = true)
    private String uid;

    private Semester sem1;
    private Semester sem2;
    private Semester sem3;
    private Semester sem4;
    private Semester sem5;
    private Semester sem6;
    private Semester sem7;
    private Semester sem8;
}
